### DISCUSSION TITLE
Meaning of google services info keys

### DISCUSSION THREAD

**Discussion by jesus-mg-ios** [firebase/firebase-ios-sdk#12961] [Q&A]
Hi, recently I downloaded a google service info plist again and I saw a mismatch between values, searching on google about the meaning of the keys didn't clear me anything, is something very confusing.

Would be great to know which keys are required for iOS and MacOS, because the new plist has more keys than before.

So if @ncooke3 could clarify all of them and say if they are required or not, and the affectation to include or not inside the configure would be great.

* CLIENT_ID
* REVERSED_CLIENT_ID
* API_KEY
* GCM_SENDER_ID
* PLIST_VERSION
* BUNDLE_ID
* PROJECT_ID
* STORAGE_BUCKET
* IS_ADS_ENABLED
* IS_ANALYTICS_ENABLED
* IS_APPINVITE_ENABLED
* IS_GCM_ENABLED
* IS_SIGNIN_ENABLED
* GOOGLE_APP_ID
* DATABASE_URL

**Comment by jesus-mg-ios** [comment]
Some info with questions unresolved by @ncooke3 , maybe it's time to clarify them and clean up unneeded keys, for now, no reading them if they are not needed https://github.com/firebase/firebase-ios-sdk/blob/31defd7e8be01dc36f7b07fee76422a4cccf33e4/docs/FirebaseOptionsPerProduct.md#L58

**Reply by ncooke3** [reply]
Thanks, jesus-mg-ios@. This is on the team's radar. Could you share a little more about cases where the `GoogleService-Info.plist` get confusing to reason about. Are you configuring your Firebase app with an options object that is configured inline as opposed to including the plist in your workspace?

**Reply by jesus-mg-ios** [reply]
```
IS_ADS_ENABLED
IS_ANALYTICS_ENABLED
IS_APPINVITE_ENABLED
IS_GCM_ENABLED
IS_SIGNIN_ENABLED
``` 
Are those needed? Because values have changed between versions. I'm trying to configure them in line with one project and the plist in another.

**Reply by ncooke3** [reply]
> Are those needed?

They shouldn't be on the latest versions of the SDK. 

>  Because values have changed between versions.

Just so I better understand where/how the values changed, what does "between versions" mean here? Different times the plist was downloaded from the Firebase console?

**Reply by jesus-mg-ios** [reply]
Yes, between downloads of the plist from the console. Some of them toggled the value

**Reply by jesus-mg-ios** [reply]
So we can assure these values for version 10.25.0 are not needed on the property list for the right SDK work. Isn't it?

**Reply by ncooke3** [reply]
That's correct.

**Reply by jesus-mg-ios** [reply]
@ncooke3 Are there any tasks related to remove those keys from firebase console plist for iOS?

**Reply by ncooke3** [reply]
Yes, improvements to the config files are being investigated across all the platforms (iOS, Android, etc.). There should be more news to come in 2025.

**Reply by paulb777** [reply]
See https://github.com/firebase/firebase-ios-sdk/pull/13587
